rest/route/agent.go

```diff
@@ -153,6 +153,35 @@ func (h *agentCheckGetPullRequestHandler) Parse(ctx context.Context, r *http.Req
 }
 
 func (h *agentCheckGetPullRequestHandler) Run(ctx context.Context) gimlet.Responder {
```

Comment:
We have MergeCommitSHA in the agent already.
https://github.com/evergreen-ci/evergreen/blob/5f7ff6c5f3eda47b655e9e5329c662d581e5000a/agent/command/git.go#L310
I see not removing this route yet in case of rollback or etc, but can we just remove the call to it on the agent-side and use MergeCommitSHA directly? Can remove a bit of this logic too.